File: src/test/java/at/chaoticbits/api/ApiTests.java
package at.chaoticbits.api;


import org.testng.Assert;
import org.testng.annotations.Test;

public class ApiTests {

    private static final String urlOK = "http://ip.jsontest.com/";
    private static final String urlFail = "fail-test";


    @Test
    public void testFetch() {
        Response response = Api.fetch(urlOK);

        Assert.assertNotNull(response);
        Assert.assertEquals(response.getStatus(), 200);
        Assert.assertNotNull(response.getBody());
    }

    @Test
    public void testFetchFail() {
        Response response = Api.fetch(urlFail);

        Assert.assertNull(response);
    }
}

File: src/test/java/at/chaoticbits/api/ApiTests.kt
package at.chaoticbits.api


import org.testng.Assert
import org.testng.annotations.Test

class ApiTests {


    companion object {
        private const val urlOK = "http://ip.jsontest.com/"
        private const val urlFail = "fail-test"
    }

    @Test
    fun testFetch() {
        val response = Api.fetch(urlOK)

        Assert.assertNotNull(response)
        Assert.assertEquals(response.status, 200)
        Assert.assertNotNull(response.body)
    }

    @Test(expectedExceptions = [(IllegalStateException::class)], expectedExceptionsMessageRegExp = "Error fetching url.*")
    fun testFetchFail() {
        val response = Api.fetch(urlFail)

        Assert.assertNull(response)
    }
}
